tests/e2e/custom_macros/mod.rs

```diff
@@ -0,0 +1,89 @@
+use indoc::indoc;
+use itertools::Itertools;
+use lsp_types::Diagnostic;
+use serde::Serialize;
+use serde_json::json;
+
+use crate::support::normalize::normalize;
+use crate::support::sandbox;
+
+#[derive(Serialize)]
+struct DiagnosticsWithUrl {
+    url: String,
+    diagnostics: Vec<Diagnostic>,
+}
+
+#[derive(Serialize)]
+struct DiagnosticsReport {
+    diagnostics: String,
+}
+
+#[test]
+fn test_custom_macro() {
```

Comment:
I'm afraid our CI times will explode with this because of Rust compilation. I wonder if we could prepublish this macro to scarbs.xyz, with precompiled binaries and use it that way. Wdyt? No need to address this in this PR, but worth deciding on now

Reply:
Yeah that's something that i want to address as a follow up. 
Let me create an issue

Reply:
https://github.com/software-mansion/cairols/issues/274